### PR TITLE
Electron | Automatic updates implemented for each release and pre-configured application for windows

### PR DESCRIPTION
**Pull request by chrismichaelps**
Automatic update for each release has been integrated, with pre-configuration for windows. It can also be configured for other platforms

# vue.config.js
Predefined settings for windows, but can be configured for other platforms.

In the **publish** section you should change:

- [x] The value of the `owner` property to the username you use on github.
- [x] The property value `repo` repository name on github.

and make value changes in the properties `appId`, and `productName`.

to see available options, check out [Electron Builder Configuration Options](https://www.electron.build/configuration/configuration)

# GH_TOKEN
You should assign an environment variable called `GH_TOKEN` with the token that github provides you. This is mandatory as it will help to apply automatic updates. For more information visit the [Auto Update](https://www.electron.build/auto-update) section for more details.

# How to release the application to github?
- [x] First, you must generate the icons that will be used as the logo of the executable icon and the application installation process.

Before generating the icons you should go to the **package.json** and change the `--input` path  where the icon is located

```json
"electron:generate-icons": "electron-icon-builder --input=./public/favicon-32x32.png --output=build --flatten",
```

then ...

```shell
yarn electron:generate-icons
```

- [x] Second, you should create the environment variable named `GH_TOKEN` and assign the value of the token that github provides you.

- [x] Then with the following command

```shell
yarn gh-publish GH_TOKEN
```

It will perform the build of the application and the release as draf. Now you should go to the repository and you will see that a release draf has created for you.

This is very useful because now the user should not go to the release section to download the new release of the application. It would only be enough to enter the application, and if there is a new version windows will show a notification to perform the update.

But yes, the user the first time should download the release from github. To see more releases the user will receive a notification to their computer to download the new version, without the need to go to github and download the release.

The documentation comes from a template that I had implemented with vue + electron, at the following [URL](https://github.com/ChrisMichaelPerezSantiago/vue-electron-boilerplate)

### CODE REVIEW

**Comment by w3cj** [comment]
Thanks for doing this! I'll take a look / merge the changes next time I stream.

File: client/src/background.js [hunk 1]
```diff
@@ -32,10 +34,27 @@ function createWindow() {
...
+    autoUpdater.checkForUpdatesAndNotify();
   }
 
+  const ses = win.webContents.session;
+
   win.on('closed', () => {
     win = null;
+    // clean storages after closing app
+    ses.clearStorageData({
+      storages: ['localstorage', 'caches', 'indexdb']
```
Comment (w3cj) [inline client/src/background.js]: Lets keep localStorage around so that the user does not have to login again.

File: client/package-lock.json [hunk 2]
```diff
@@ -935,7 +935,6 @@
       "version": "7.9.6",
```
Comment (w3cj) [inline client/package-lock.json]: There are a lot of changes in this file! Can you please remove it from the pull request, and I will generate it locally. Thanks!

**Review by w3cj — CHANGES_REQUESTED** [review]
See comments. Thanks for doing this!

File: client/vue.config.js [hunk 3]
```diff
@@ -1,3 +1,34 @@
 module.exports = {
   lintOnSave: false,
+  productionSourceMap: false,
```
Comment (w3cj) [inline client/vue.config.js:12]: Please merge with the recent changes to vue.config.js

File: client/src/background.js (hunk 1 above)
Comment (chrismichaelps) [reply client/src/background.js]: Changes: The code was documented.

File: client/package-lock.json (hunk 2 above)
Comment (chrismichaelps) [reply client/package-lock.json]: client/package.lock.json file removed.

File: client/vue.config.js (hunk 3 above)
Comment (chrismichaelps) [reply client/vue.config.js:12]: electronBuilder plugin option added to the vue.config.js file

**Comment by chrismichaelps** [comment]
Hello @w3cj ,

I have already made the changes indicated.
And apologies for leaving the `package-lock.js` file, I already removed it as you indicated.

**Comment by w3cj** [comment]
Thanks, but the package-lock.json is now removed! Can you make a commit that does not include any changes / deletions to the package-lock.json?

**Pull request closed without merging**